### PR TITLE
Add configuration doc

### PR DESCRIPTION
**Pull request by yiheng**
## What changes were proposed in this pull request?
Add a configuration page containing all java properties controling bigdl behavior.

## How was this patch tested?
No need test as document change. maybe later can run build doc website command to test.

## Related links or issues (optional)
https://github.com/intel-analytics/BigDL/issues/1190

### CODE REVIEW

File: docs/docs/UserGuide/configuration.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# BigDL Configuration
+BigDL uses Java properties to control its behavior. Here's the list of
+these properties.
```
Comment (jason-dai) [inline docs/docs/UserGuide/configuration.md]: How to set the properties?

File: docs/docs/UserGuide/configuration.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Default value is **Current_Working_Directory/bigdl.log**. Where is the redirecting log.
+
+* bigdl.utils.LoggerFilter.enableSparkLog
+
+Default value is **true**. Enable redirecting Spark logs to logFile
+
+## Mode
+* bigdl.localMode
+
+Default value is **false**. Whether BigDL is running on a local JVM.
```
Comment (jason-dai) [inline docs/docs/UserGuide/configuration.md]: Whether BigDL is running as a local Java/Scala program

File: docs/docs/UserGuide/configuration.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+* bigdl.engineType
+
+Default value is **mklblas**. Specify which kind of native library to speed up the computing.
+
+## Multi-threading
+
+* bigdl.coreNumber
+
+Default value is **half of the virtual core number**. How many core BigDL use on your machine. If
```
Comment (jason-dai) [inline docs/docs/UserGuide/configuration.md]: It is only used when bigdl.localMode is set to true

File: docs/docs/UserGuide/configuration.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+Default value is **4**. Thread pool size for syncing parameter between executors.
+
+
+## Distributed Training
+* bigdl.network.nio
+
+Default value is **true**. Whether use NIO as BlockManager backend in Spark 1.5. If it is set to
+false, user can specify spark.shuffle.blockTransferService to change the BlockManager backend.
+
+Note this only take affect on Spark 1.5.
```
Comment (jason-dai) [inline docs/docs/UserGuide/configuration.md]: It is only used when running on Spark 1.5

File: docs/docs/UserGuide/configuration.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+Default value is **true**. Enable redirecting Spark logs to logFile
+
+## Mode
+* bigdl.localMode
+
+Default value is **false**. Whether BigDL is running on a local JVM.
+
+* bigdl.engineType
+
+Default value is **mklblas**. Specify which kind of native library to speed up the computing.
```
Comment (dding3) [inline docs/docs/UserGuide/configuration.md]: Looks like currently we only support mklblas. Do we need clarify that or remove this config here?

File: docs/docs/UserGuide/configuration.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+## Multi-threading
+
+* bigdl.coreNumber
+
+Default value is **half of the virtual core number**. How many core BigDL use on your machine. If
+hyper thread is enabled on your machine, DO NOT set it larger than half of the virtual core number.
+
+* bigdl.Parameter.syncPoolSize
+
```
Comment (dding3) [inline docs/docs/UserGuide/configuration.md]: move "bigdl.Parameter.syncPoolSize" under "Distributed Training" section?

File: docs/docs/UserGuide/configuration.md (hunk 5 above)
Comment (jason-dai) [inline docs/docs/UserGuide/configuration.md]: Yes, we can remove it

File: docs/docs/UserGuide/configuration.md (hunk 6 above)
Comment (yiheng) [inline docs/docs/UserGuide/configuration.md]: Then there's only one under multi-threading section...

File: docs/docs/UserGuide/configuration.md (hunk 1 above)
Comment (yiheng) [inline docs/docs/UserGuide/configuration.md]: addressed.

**Pull request merged**